CI: TestBPF fails with "-bpf-test-path is a required flag" in runtime DatapathPrivilegedUnitTests

[09:02:19]     bpf_test.go:56: -bpf-test-path is a required flag
	 [09:02:19] --- FAIL: TestBPF (0.00s)
	 [09:02:19] FAIL
	 [09:02:19] coverage: 6.2% of statements
	 [09:02:19] FAIL	github.com/cilium/cilium/test/bpf_tests	0.037s

Started failing consistently on `master` after https://github.com/cilium/cilium/pull/22043

Latest failure on `master`: https://jenkins.cilium.io/job/cilium-master-runtime-kernel-net-next/5238/

--- a/.github/workflows/lint-bpf-checks.yaml
+++ b/.github/workflows/lint-bpf-checks.yaml
@@ -131,9 +131,10 @@ jobs:
         with:
           persist-credentials: false
           fetch-depth: 0
-      - name: Run BPF tests with code coverage reporting
+      - name: Run BPF tests
         run: |
-          make -C test run_bpf_tests COVER=1 || (echo "Run 'make -C test run_bpf_tests COVER=1' locally to investigate failures"; exit 1)
+          # TODO: re-enable coverage reporting, see https://github.com/cilium/cilium/issues/22088
+          make -C test run_bpf_tests || (echo "Run 'make -C test run_bpf_tests' locally to investigate failures"; exit 1)
       - name: Archive code coverage results
         uses: actions/upload-artifact@v3
         with:

--- a/test/Makefile
+++ b/test/Makefile
@@ -85,4 +85,4 @@ endif
 
 run_bpf_tests:
 	$(QUIET)$(MAKE) -C ../bpf/tests all
-	$(QUIET)$(GO) test bpf_tests/*.go -exec sudo -bpf-test-path $(ROOT_DIR)/bpf/tests $(BPF_TEST_FLAGS)
+	$(QUIET)$(GO) test ./bpf_tests -exec sudo -bpf-test-path $(ROOT_DIR)/bpf/tests $(BPF_TEST_FLAGS)

--- a/test/bpf_tests/bpf_test.go
+++ b/test/bpf_tests/bpf_test.go
@@ -36,7 +36,6 @@ import (
 	"github.com/cilium/cilium/pkg/byteorder"
 	"github.com/cilium/cilium/pkg/datapath/link"
 	"github.com/cilium/cilium/pkg/monitor"
-	"github.com/cilium/cilium/pkg/testutils"
 )
 
 var (
@@ -50,10 +49,8 @@ var (
 )
 
 func TestBPF(t *testing.T) {
-	testutils.PrivilegedTest(t)
-
 	if testPath == nil || *testPath == "" {
-		t.Fatal("-bpf-test-path is a required flag")
+		t.Skip("Set -bpf-test-path to run BPF tests")
 	}
 
 	entries, err := os.ReadDir(*testPath)

